ui-tests/src/test/java/org/openmrs/reference/page/ManageUserAccountPage.java

```diff
@@ -0,0 +1,148 @@
+    private static final By GENDER_FEMALE = By.cssSelector("#adminui-gender-1-field");
+    private static final By ADD_USER_ACCOUNT = By.id("adminui-addUserAccount");
+    private static final By USER_NAME = By.cssSelector("#adminui-username-field");
+    private static final By PRIVILEGE_LEVEL = By.cssSelector("#adminui-privilegeLevel-field");
+    private static final By PASSWORD = By.cssSelector("#adminui-password-field");
+    private static final By CONFIRM_PASSWORD = By.cssSelector("#adminui-confirmPassword-field");
+    private static final By ADD_PROVIDER_ACCOUNT = By.id("adminui-addProviderAccount");
+    private static final By IDENTIFIER = By.cssSelector("#adminui-identifier-field");
+    private static final By PROVIDER_ROLE = By.cssSelector("#adminui-providerRole-field");  
+    private static final By EDIT_USER_ACCOUNT = By.cssSelector("i.icon-pencil.edit-action");
+    private static final By ADMINISTER_SYSTEM = By.id("adminui-capabilities-Application: Administers System");
+    private static final By CONFIGURE_FORMS = By.id("adminui-capabilities-Application: Configures Forms");
+    private static final By CAPTURE_VITALS = By.id("adminui-capabilities-Application: Enters Vitals");
+    private static final By RECORD_ALLERGIES = By.id("adminui-capabilities-Application: Records Allergies");
+    private static final By CONFIGURE_METADATA = By.id("adminui-capabilities-Application: Configures Metadata");
+    private static final By SUPER_USER_PRIVILEGES = By.id("adminui-capabilities-Application: Has Super User Privileges");
+    private static final By REGISTER_PATIENT = By.id("adminui-capabilities-Application: Registers Patients");
+    private static final By SCHEDULE_APPOINTMENTS = By.id("adminui-capabilities-Application: Schedules And Overbooks Appointments"); 
+    private static final By PATIENT_SUMMARY = By.id("adminui-capabilities-Application: Uses Patient Summary");  
+    private static final By SAVE_BUTTON = By.id("save-button");
+
+    public ManageUserAccountPage(Page page) {
+        super(page);
+    }
+
+    @Override
+    public String getPageUrl() {
+        return URL;
+    }
+
+    public void clickOnAddUser() {
+        clickOn(ADD_NEW_ACCOUNT);
+    }
+    
+    public void searchUser(String username) {
+        setTextToFieldNoEnter(SEARCH_USER, username);
+    }
+    
+    public void enterPersonalDetails(String familyName, String givenName) {
+        findElement(FAMILY_NAME).clear();
+        findElement(FAMILY_NAME).sendKeys(familyName); 
+        findElement(GIVEN_NAME).clear();
+        findElement(GIVEN_NAME).sendKeys(givenName);
+    }
+    
+    public void selectGender() {
+        clickOn(GENDER_FEMALE);
+    }
+
+    public void clickOnAddUserAccount() {
+        clickOn(ADD_USER_ACCOUNT);
+    }
+   
+    public void enterUsername(String username) {
+        findElement(USER_NAME).clear();
+        findElement(USER_NAME).sendKeys(username);
```

Comment:
I'm of two minds about this pattern. On the one hand, its necessary to reliably do what the test needs to do. On the other hand, if there's data pre-populated into that field, that seems like an issue that should be flagged. Thoughts?

Reply:
Here is my thought on the epic:
Reliably making the pattern to directly do what the logic is supposed to perform would be ok, however ,on the the other hand there could be data pre-populated in the field and invoking  clear() method makes things right using the same pattern. 
For-example when editing user account details the same pattern will cater for such scenarios than again writing another pattern that can edit the data, so I thought catching two birds in one net would be cool than using two nets to catch two birds. 

Echoing on the first thought invoking clear() method has no harm at all but does the magic for the second  thought!